src/Controller/TrickController.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Trick;
+use App\Form\TrickType;
+use DateTime;
+use Doctrine\Common\Persistence\ObjectManager;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+class TrickController extends AbstractController
+{
+    /**
+     * @Route("/trick/{id}", name="trick_show_id", requirements={"id": "\d+"})
+     */
+    public function show(Trick $trick)
+    {
+        return $this->render('trick/trick.html.twig', [
+            'trick' => $trick
+        ]);
+    }
+
+    /**
+     * @Route("/create-trick", name="create_trick")
+     * @Route("/edit-trick/{id}", name="edit_trick", requirements={"id": "\d+"})
+     */
+    public function createOrEdit(Request $request, ObjectManager $manager, Trick $trick = null)
+    {
+        $trick = $trick ?? new Trick();
+
+        $form = $this->createForm(TrickType::class, $trick);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            if (!$trick->getId()) {
+                $trick->setCreatedAt(new DateTime);
```

Comment:
Move this assignation is the Trick constructor

Reply:
Better: I use Doctrine PrePersist annotation now!